### PR TITLE
Add 1.12.1 torch compatibility in sum method

### PR DESCRIPTION
**Pull request by PanchenkoYehor**
The current code needs at least 1.13 torch which is not very convenient, because 1.13 torch deprecated CUDA 10.2 and patch 1.13 contains several major updates. There is a high chance that not every project is ready to update their environments and CUDA drivers to use 0.3.1 trl version

### CODE REVIEW

**Comment by PanchenkoYehor** [comment]
The relevant open issue that currently resolves by updating environment #183

**Review by younesbelkada — COMMENTED** [review]
Hi @PanchenkoYehor 
Thanks a lot for the fix!
Can you run the styling checks? `make style` + `make quality`
Thanks!

File: trl/core.py [hunk 1]
```diff
@@ -106,14 +104,17 @@ def whiten(values, shift_mean=True):
 
 def masked_mean(values, mask, axis=None):
     """Compute mean of tensor with a masked values."""
-    return (values * mask).sum(axis=axis) / mask.sum(axis=axis)
+    try:
+        return (values * mask).sum(axis=axis) / mask.sum(axis=axis)
+    except:
+        return (values * mask).sum() / mask.sum()
```
[the remark this answers is not in the document]
Comment (PanchenkoYehor) [reply trl/core.py:110]: Good point, I will change it

**Comment by lvwerra** [comment]
Looks great, thanks for fixing!

**Review by lvwerra — APPROVED** [review]

File: examples/summarization/scripts/reward_summarization.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 from transformers.utils import PaddingStrategy
 
 
+
```
Comment (younesbelkada) [inline examples/summarization/scripts/reward_summarization.py:19]:
```suggestion
```

File: trl/core.py [hunk 3]
```diff
@@ -26,7 +26,6 @@
 except ImportError:
     from collections import Mapping
```
Comment (younesbelkada) [inline trl/core.py:28]:
```suggestion


```

**Review by younesbelkada — APPROVED** [review]
Thanks a lot for this!

**Pull request merged**